platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformInputArtifactIntegrationTest.groovy

```diff
@@ -348,20 +361,29 @@ class ArtifactTransformInputArtifactIntegrationTest extends AbstractDependencyRe
         when:
         succeeds(":a:resolve")
 
-        then: // have already seen these artifacts before, but the transform outputs have been overwritten
+        then:
         result.assertTasksNotSkipped(":b:producer", ":a:resolve")
-        transformed("b-dir")
+        // have already seen these artifacts before...
+        if (incremental) {
```

Comment:
💅 Would it make sense to split this test in two?

Reply:
Like into two test cases? I think that would make things worse. Incrementality should have little to do with what this test is about, we are only checking both cases to be extra sure. For that reason I think it's better to have the `if` here. To be honest, I think it would also be okay to remove this last `when:`/`then:` section.